server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -124,7 +116,7 @@ public CompressedXContent(byte[] data) throws IOException {
     }
 
     public CompressedXContent(String str) throws IOException {
-        this(new BytesArray(new BytesRef(str)));
+        this(new BytesArray(str.getBytes(StandardCharsets.UTF_8)));
```

Comment:
This should be much more efficient here as we mainly use it on strings that are internally coded as UTF-8 already (mapping update requests) so we just do a simple `byte[]` copy here instead of creating an oversized array and doing char by char work in the `BytesRef` code